### DISCUSSION TITLE
Possibility for compiler warning

### DISCUSSION THREAD

**Discussion by wrestang** [gabime/spdlog#2330] [Q&A] [answered]
Is there any way to have a compile time warning when too many `{}` are added to a log line?  I'm not sure if this would be more of a fmt library issue but, when I have a log line like:
`spdlog::debug("{} this is my log {}", "a");`
Instead of simply logging out an exact `{}`, the entire log line is dropped.  I just so happened to find some cases of this, and I'm not sure how widespread it might be, and it is nearly impossible to search for this by hand.  If there was some way to enable a compile time warning, that would be most excellent.
I've opened this as a Q&A discussion, because it may already be possible, and I just don't know it yet.

**Comment by tt4g** [comment, marked as answer]
The `FMT_STRING` macro is provided by fmt library (see https://fmt.dev/latest/index.html#safety).
This feature implemented by #1574.

Example:

```cpp
spdlog::debug(FMT_STRING("{} this is my log {}"), "a");
```

Related #1510

**Reply by wrestang** [reply]
Thanks, that does look like a workaround, despite the fact that implementing would take a decent script to have it push out to a large number of files/log instances.
It does also seem to indicate it may be possible to do that call automatically via spdlog based off of a compile time argument, assuming that spdlog does a direct `fmt::format` call like that. Of course, I could also be the only person in the world that even wants that feature.
